cmd/resource.go

```diff
@@ -23,107 +22,129 @@ import (
 var validateResourceName = utils.ValidatorFactory.NewFromRegex(`^[a-zA-Z0-9][a-zA-Z0-9_\-\.]+$`,
 	`invalid name (can only contain characters A-Z (in either case), 0-9, "-", "_" or "." and must start with an alphanumeric character)`)
 
-func resourceCommand(l log.Logger, conf config.Optimus, datastoreRepo models.DatastoreRepo, datastoreSpecsFs map[string]map[string]afero.Fs) *cli.Command {
+func resourceCommand() *cli.Command {
+	var configFilePath string
+	conf := &config.ClientConfig{}
+	l := initDefaultLogger()
+
 	cmd := &cli.Command{
 		Use:   "resource",
 		Short: "Interact with data resource",
 		Annotations: map[string]string{
 			"group:core": "true",
 		},
 	}
-	cmd.AddCommand(createResourceSubCommand(l, conf, datastoreSpecsFs, datastoreRepo))
+	cmd.PersistentPreRunE = func(cmd *cli.Command, args []string) error {
+		// TODO: find a way to load the config in one place
+		var err error
+
+		conf, err = config.LoadClientConfig(configFilePath)
+		if err != nil {
+			return err
+		}
+		l = initClientLogger(conf.Log)
+
+		return nil
+	}
+
+	cmd.AddCommand(createResourceSubCommand(l, conf))
 	return cmd
 }
 
-func createResourceSubCommand(l log.Logger, conf config.Optimus, datastoreSpecFs map[string]map[string]afero.Fs, datastoreRepo models.DatastoreRepo) *cli.Command {
+func createResourceSubCommand(l log.Logger, conf *config.ClientConfig) *cli.Command {
 	cmd := &cli.Command{
 		Use:     "create",
 		Short:   "Create a new resource",
 		Example: "optimus resource create",
-		RunE: func(cmd *cli.Command, args []string) error {
-			namespace, err := askToSelectNamespace(l, conf)
-			if err != nil {
-				return err
-			}
-			availableStorer := []string{}
-			for _, s := range datastoreRepo.GetAll() {
-				availableStorer = append(availableStorer, s.Name())
-			}
-			var storerName string
-			if err := survey.AskOne(&survey.Select{
-				Message: "Select supported datastores?",
-				Options: availableStorer,
-			}, &storerName); err != nil {
-				return err
-			}
-			repoFS, ok := datastoreSpecFs[namespace.Name][storerName]
-			if !ok {
-				return fmt.Errorf("unregistered datastore, please use configuration file to set datastore path")
-			}
+	}
 
-			// find requested datastore
-			availableTypes := []string{}
-			datastore, _ := datastoreRepo.GetByName(storerName)
-			for dsType := range datastore.Types() {
-				availableTypes = append(availableTypes, dsType.String())
-			}
-			resourceSpecRepo := local.NewResourceSpecRepository(repoFS, datastore)
-
-			// find resource type
-			var resourceType string
-			if err := survey.AskOne(&survey.Select{
-				Message: "Select supported resource type?",
-				Options: availableTypes,
-			}, &resourceType); err != nil {
-				return err
-			}
-			typeController := datastore.Types()[models.ResourceType(resourceType)]
+	cmd.RunE = func(cmd *cli.Command, args []string) error {
+		dsRepo := models.DatastoreRegistry
```

Comment:
lets check on whether we can improve how the dsRepo initialization can be improved too

Reply:
It is initialized `github.com/odpf/optimus/ext/datastore/bigquery`. I think it's good like this as it is initialized once and used multiple times. With the current approach, we don't need to initialized it multiple times if required, nor do we need to pass it as arguments. We just call it as it is `models.DatastoreRegistry`.

Reply:
Fyi, initialize `PluginRegistry` is different with `DatastoreRegistry`. `PluginRegistry` can only be loaded by calling plugin.Initialize(), while `DatastoreRegistry` can be initialize by calling blank import to trigger init function on `github.com/odpf/optimus/ext/datastore/bigquery`

as of now, let's keep same. we can improve initialization later as part of client command refactoring